doc/source/whatsnew/v2.2.0.rst

```diff
@@ -169,6 +169,7 @@ Bug fixes
 ~~~~~~~~~
 - Bug in :class:`AbstractHolidayCalendar` where timezone data was not propagated when computing holiday observances (:issue:`54580`)
 - Bug in :class:`pandas.core.window.Rolling` where duplicate datetimelike indexes are treated as consecutive rather than equal with ``closed='left'`` and ``closed='neither'`` (:issue:`20712`)
+- Bug in :func:`is_all_strings` while checking object array with no elements is of the string dtype (:issue:`54661`)
```

Comment:
is the bug in is_all_strings or is_string_dtype?

Reply:
I think, the bug is in `is_all_strings`, but we don’ t have this function in api and users see the bug while calling `is_string_dtype`. That is why I replaced Bug in :func:`is_all_strings` with: Bug in :func:`is_string_dtype`.

Looks like we don’t have a test for is_all_strings, do we need one?

Reply:
i meant the user-facing bug.  in the docs we generally only reference public functions/methods

Reply:
okay, I removed from `doc/source/whatsnew/v2.2.0.rst` my line.

Reply:
@jbrockmendel, I updated the PR. Could you please take a look at it.

Reply:
Yeah, `is_string_dtype` is a public function, I added the note to `v2.1.1.rst`. I think my change is a minor and we can put it in the version 2.1.1